gen3/bin/kube-setup-argo.sh

```diff
@@ -75,10 +121,35 @@ EOF
     fi
 
     gen3_log_info "Creating s3 creds secret in argo namespace"
-    g3kubectl create secret -n argo generic argo-s3-creds --from-literal=AccessKeyId=$(echo $secret  | jq -r .AccessKey.AccessKeyId) --from-literal=SecretAccessKey=$(echo $secret  | jq -r .AccessKey.SecretAccessKey) --from-literal=bucketname=${bucketName}
+    if [[ -z $internalBucketName ]]; then
+      g3kubectl create secret -n argo generic argo-s3-creds --from-literal=AccessKeyId=$(echo $secret  | jq -r .AccessKey.AccessKeyId) --from-literal=SecretAccessKey=$(echo $secret  | jq -r .AccessKey.SecretAccessKey) --from-literal=bucketname=${bucketName}
+    else
+      g3kubectl create secret -n argo generic argo-s3-creds --from-literal=AccessKeyId=$(echo $secret  | jq -r .AccessKey.AccessKeyId) --from-literal=SecretAccessKey=$(echo $secret  | jq -r .AccessKey.SecretAccessKey) --from-literal=bucketname=${bucketName} --from-literal=internbucketname=${internalBucketName}
+    fi
   else 
     gen3_log_info "Argo S3 setup already completed" 
   fi
+
+  gen3_log_info "Adding indexd and fence creds to argo namespace"
+  for serviceName in indexd fence; do
```

Comment:
we don't need fence creds; we need admin user creds for indexd; previously we used the same fence-indexd admin user creds